plugins/usbdmx/AsyncUsbTransceiverBase.cpp

```diff
@@ -67,62 +102,83 @@ bool AsyncUsbTransceiverBase::Init() {
 }
 
 void AsyncUsbTransceiverBase::CancelTransfer() {
-  if (!m_transfer) {
-    return;
+  m_suppress_continuation = true;
+
+  inflight_set::iterator i;
+  for (i = m_inflight.begin(); i != m_inflight.end(); ++i) {
+    int ret = m_adaptor->CancelTransfer(*i);
+    /* We ignore NOT_FOUND errors here because transfer cancellation
+     * naturally races with completion.
+     *
+     * This error simply means the transfer is not in the "in progress"
+     * state anymore.
+     *
+     * Even if we were to check for this before cancelling it could still
+     * happen that the transfer completes between the check and the
+     * cancellation taking effect.
+     */
+    if (ret != 0 && ret != LIBUSB_ERROR_NOT_FOUND) {
+      OLA_WARN << "libusb_cancel_transfer returned "
+               << m_adaptor->ErrorCodeToString(ret);
+    }
```

Comment:
I wondered if we should have some debug here, but I guess if we're only calling cancel during shutdown/disconnect it's not very exciting.

Reply:
What sort of debug prints are you thinking of beyond the error string?

Reply:
Just logging the LIBUSB_ERROR_NOT_FOUND, but perhaps at OLA_DEBUG level, so we can see all the failures.

Reply:
Done, log level now depends on where the error is NOT_FOUND or not.